### PR TITLE
Lera/features/teacher courses endpoints

### PR DESCRIPTION
**Pull request by arstrybuh257**

### CODE REVIEW

File: UniAtHome/UniAtHome.BLL/Util/MappingProfile.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using UniAtHome.DAL.Entities;
+
+namespace UniAtHome.BLL.Util
+{
+    public class MappingProfile: Profile
+    {
+        public MappingProfile()
+        {
+            CreateMap<Course, CourseDTO>();
+            CreateMap<CourseDTO, Course>();
```
Comment (arstrybuh257) [inline UniAtHome/UniAtHome.BLL/Util/MappingProfile.cs]: Please, use .ReverseMap()

File: UniAtHome/UniAtHome.DAL/Repositories/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+using UniAtHome.DAL.Entities;
+using UniAtHome.DAL.Interfaces;
+
+namespace UniAtHome.DAL.Repositories
+{
+    public class Repository<T> : IRepository<T> where T: BaseEntity
```
Comment (arstrybuh257) [inline UniAtHome/UniAtHome.DAL/Repositories/Repository.cs:8]: Maybe, we shouldn't use generic repository. As for me, it is not convenient way. What do you think?

File: UniAtHome/UniAtHome.BLL/Util/MappingProfile.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using AutoMapper;
+using UniAtHome.BLL.DTOs;
+using UniAtHome.DAL.Entities;
+
+namespace UniAtHome.BLL.Util
```
Comment (arstrybuh257) [inline UniAtHome/UniAtHome.BLL/Util/MappingProfile.cs]: Please, put automapper configuration in ./Configuration folder in the WebAPI project

Comment (EdShel) [inline UniAtHome/UniAtHome.BLL/Util/MappingProfile.cs]: Don't name folders as Util, Extensions etc: it does not make any sense. Give meaningful names

**Review by EdShel — CHANGES_REQUESTED** [review]

File: UniAtHome/UniAtHome.DAL/Repositories/Repository.cs (hunk 2 above)
Comment (lesloress) [reply UniAtHome/UniAtHome.DAL/Repositories/Repository.cs:8]: many functions are common for all the repos (like getbyid, remove, add, etc), so it`s neccessary

**Pull request merged**